src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnectionString.cs

```diff
@@ -357,11 +359,20 @@ internal SqlConnectionString(string connectionString) : base(connectionString, G
 
         // This c-tor is used to create SSE and user instance connection strings when user instance is set to true
         // BUG (VSTFDevDiv) 479687: Using TransactionScope with Linq2SQL against user instances fails with "connection has been broken" message
-        internal SqlConnectionString(SqlConnectionString connectionOptions, string dataSource, bool userInstance) : base(connectionOptions)
+        internal SqlConnectionString(SqlConnectionString connectionOptions, string dataSource, bool userInstance, bool? setEnlistValue) : base(connectionOptions)
```

Comment:
Change to constructor brings in full support for enlist keyword. After this change SqlConnectionString type doesn't need any work for enlist keyword.